### PR TITLE
The correct PluginConfig.Args is not passed to the corresponding PluginFactory in kube-scheduler when multiple PluginConfig items are defined

### PR DESCRIPTION
**Pull request by everpeace**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

/sig scheduling

`PluginConfig.Args` is not passed as expected if `PluginConfig` has multiple items.  It is because the type of `PluginConfig.Args` is not pointer type.

**Which issue(s) this PR fixes**:

It fixes the above issue so that `NewFramework` correctly initialize `PluginConfig.Args` as expected.

**Special notes for your reviewer**:

the PR did not change api(`KubeSchedulerConfiguration`) at all but fixes inside of kube-scheduler (just allocating new variable to capture the correct `PluginConfig.Args`)

**Does this PR introduce a user-facing change?**:

```release-note
fixed an issue that the correct PluginConfig.Args is not passed to the corresponding PluginFactory in kube-scheduler when multiple PluginConfig items are defined.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/v1alpha1/framework_test.go
Comment (draveness) [inline pkg/scheduler/framework/v1alpha1/framework_test.go]: nit: \n

**Review by draveness — COMMENTED** [review]
/ok-to-test
/priority important-soon

File: pkg/scheduler/framework/v1alpha1/framework_test.go
Comment (draveness) [inline pkg/scheduler/framework/v1alpha1/framework_test.go]: nit: pluginConfigs

**Review by draveness — COMMENTED** [review]
Generally looks good, just a few nits:

File: pkg/scheduler/framework/v1alpha1/framework_test.go
Comment (everpeace) [reply pkg/scheduler/framework/v1alpha1/framework_test.go]: fixed in https://github.com/kubernetes/kubernetes/pull/82483/commits/b3a82e3b93af38c478b6e65145e24e4730ef7882

File: pkg/scheduler/framework/v1alpha1/framework_test.go
Comment (everpeace) [reply pkg/scheduler/framework/v1alpha1/framework_test.go]: fixed in https://github.com/kubernetes/kubernetes/pull/82483/commits/b3a82e3b93af38c478b6e65145e24e4730ef7882

**Comment by everpeace** [comment]
@draveness Thanks for your review.  I fixed my code as your comments.

**Review by draveness — COMMENTED** [review]
Thanks @everpeace, looks good, please squash the commits.

**Comment by everpeace** [comment]
Thanks.  Squashed and pushed.

**Review by draveness — COMMENTED** [review]
/lgtm

Thanks!

File: pkg/scheduler/framework/v1alpha1/framework_test.go
Comment (ahg-g) [inline pkg/scheduler/framework/v1alpha1/framework_test.go]: we can use DecodeInto (exist in registry.go) instead of defining a new function for decoding.

File: pkg/scheduler/framework/v1alpha1/framework_test.go
Comment (ahg-g) [inline pkg/scheduler/framework/v1alpha1/framework_test.go]: nit: I would name it injArgs instead of injUnknown

**Review by ahg-g — COMMENTED** [review]
Good catch!

File: pkg/scheduler/framework/v1alpha1/framework.go
Comment (Huang-Wei) [inline pkg/scheduler/framework/v1alpha1/framework.go]: IMO `DeepCopy()` can be eliminated by:

```go
	for i := range args {
		p := args[i]
		pc[p.Name] = &p.Args
	}
```

or

```go
	for _, p := range args {
		p := p
		pc[p.Name] = &p.Args
	}
```

**Review by Huang-Wei — COMMENTED** [review]
/hold

Calling `DeepCopy()` seems to be unnecessary.

File: pkg/scheduler/framework/v1alpha1/framework.go
Comment (ahg-g) [reply pkg/scheduler/framework/v1alpha1/framework.go]: strange, I thought a left a comment to that affect, I probably forgot to click the submit button!

Comment (draveness) [reply pkg/scheduler/framework/v1alpha1/framework.go]: I prefer the first approach

```go
	for i := range args {
		p := args[i]
		pc[p.Name] = &p.Args
	}
```

**Comment by everpeace** [comment]
I fixed as reviewers' comments.  After being approved, I will squash my commits.
If it is ok to squash at this time, please tell me so 🙇

**Review by draveness — COMMENTED** [review]
/lgtm

File: pkg/scheduler/framework/v1alpha1/framework.go
Comment (ahg-g) [reply pkg/scheduler/framework/v1alpha1/framework.go]: or you could do to save a line :)

```
for p, i := range args {
	pc[p.Name] = &args[i].Args
}
```

**Comment by ahg-g** [comment]
Please squash the comments.

**Comment by draveness** [comment]
Please cancel the hold after squash commits

Regards,
Draven

**Comment by everpeace** [comment]
Squashed.

/hold cancel

**Review by draveness — APPROVED** [review]
/lgtm

Thanks!

**Pull request merged**
